src/jsMain/kotlin/net/kyori/adventure/webui/js/StyleButtons.kt

```diff
@@ -0,0 +1,206 @@
+package net.kyori.adventure.webui.js
+
+import kotlinx.browser.document
+import org.w3c.dom.CustomEvent
+import org.w3c.dom.CustomEventInit
+import org.w3c.dom.HTMLButtonElement
+import org.w3c.dom.HTMLDivElement
+import org.w3c.dom.HTMLInputElement
+import org.w3c.dom.HTMLTextAreaElement
+import org.w3c.dom.asList
+
+// https://iro.js.org/colorPicker_api.html
+private external interface ColorPicker {
+    // https://iro.js.org/guide.html#color-picker-events
+    fun <T> on(event: String, callback: (T) -> Unit)
+    var color: Color
+}
+
+// https://iro.js.org/color_api.html
+private external interface Color {
+    var hexString: String
+}
+
+// Copied out of adventure; can't use adventure api directly because this is javascript
+private val NAMED_COLORS = mapOf(
+    "#000000" to "black",
+    "#0000aa" to "dark_blue",
+    "#00aa00" to "dark_green",
+    "#00aaaa" to "dark_aqua",
+    "#aa0000" to "dark_red",
+    "#aa00aa" to "dark_purple",
+    "#ffaa00" to "gold",
+    "#aaaaaa" to "gray",
+    "#555555" to "dark_gray",
+    "#5555ff" to "blue",
+    "#55ff55" to "green",
+    "#55ffff" to "aqua",
+    "#ff5555" to "red",
+    "#ff55ff" to "light_purple",
+    "#ffff55" to "yellow",
+    "#ffffff" to "white",
+)
+
+public data class StyleTag(val beforeCursor: String, val placeholder: String, val beforeSel: String, val afterSel: String) {
+    public constructor(opening: String, closing: String) : this("", "", opening, closing)
+    public constructor(simple: String) : this("", "", "<$simple>", "</$simple>")
+}
+
+public val STYLE_WRAPPERS: Map<String, StyleTag> = mapOf(
+    "bold" to StyleTag("b"),
```

Comment:
Minor nitpick, but as the PR needs rebasing I thought I'd mention it. These should all be the explicit ( eg `<bold>`) versions of the tags in order to match the explicit tags for colour as well.

Reply:
the justification was that while things like `<b>` might be familiar to some people (the icon for bold is just the letter **B**, after all), `<c>` isn't *as* clear, though I can change it for consistency now and add an option later as discussed above